docs/roslyn_best_practices.md

```diff
@@ -0,0 +1,181 @@
+# Roslyn Best Practices
+
+We use [Roslyn Analyzers](https://docs.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview) highlight areas of code that will need to be refactored. By default our goal is to pair these Analyzers with Code Fixers to automate as much of the upgrade workflow as possible.
+
+We support the following types of fixes:
```

Comment:
What's this list? Maybe more of an example list of things that can be done?

Reply:
> Maybe more of an example list of things that can be done?

Yes. Seems we should reword this to clarify. How about `Example of changes that we can perform include:`